### PR TITLE
Adds toEppoString representation of EppoValue; handles (FF-2345)

### PR DESCRIPTION
**Pull request by leoromanovsky**
New test was introduced in: https://github.com/Eppo-exp/sdk-test-data/pull/25

To perform comparison between different types we need a consistent representation across SDKs.

## description

We decided on a common protocol for comparing stringified numbers, integers and decimals. This PR implements that behavior in `toEppoString`

## questions for reviews

some TODOs in-line - given the interface of swift I need to handle `throw` or allow the exception to re-bubble up.

### CODE REVIEW

File: Tests/eppo/EppoValueTests.swift [hunk 1]
```diff
@@ -68,6 +68,23 @@ class EppoValueTests: XCTestCase {
         XCTAssertFalse(arrayValue1 == arrayValueDifferent, "Arrays with different elements or counts should not be equal")
     }
 
+    func testToString() {
```
Comment (leoromanovsky) [inline Tests/eppo/EppoValueTests.swift:71]: same tests as in: https://github.com/Eppo-exp/python-sdk/pull/51

missing anything?

File: Sources/eppo/dto/EppoValue.swift [hunk 2]
```diff
@@ -152,21 +152,52 @@ public class EppoValue : Decodable, Equatable {
...
+            } else {
+                // todo: should this throw?
+                return "null"
+            }
+        case .Null:
+            return "null"
         }
+    }
 
+    public func toHashedString() -> String {
```
Comment (leoromanovsky) [inline Sources/eppo/dto/EppoValue.swift:194]: this function is used for the assignment logging to keep a consistent cache entry size (https://github.com/Eppo-exp/eppo-ios-sdk/blob/main/Sources/eppo/AssignmentCache.swift#L30). it's probably extraneous and I could just use the new function in this PR. Thoughts welcome.

File: Sources/eppo/dto/EppoValue.swift [hunk 3]
```diff
@@ -152,21 +152,52 @@ public class EppoValue : Decodable, Equatable {
...
+                    // If the double value is an integer, return as integer string
+                    return String(format: "%.0f", doubleVal)
+                } else {
+                    // Otherwise, return the double as is
+                    return String(doubleVal)
+                }
+            } else {
+                // todo: should this throw?
+                return "null"
+            }
```
Comment (schmit) [inline Sources/eppo/dto/EppoValue.swift:171]: When would the self.getDoubleValue fail?

File: Sources/eppo/dto/EppoValue.swift [hunk 4]
```diff
@@ -152,21 +152,52 @@ public class EppoValue : Decodable, Equatable {
...
+            // todo: should this throw?
+            // it should not since we checked the type is a string above
+            // but to be true to the type system we should throw here
+            if let value = try? self.getStringValue() {
+                return value
+            } else {
+                // todo: should this throw?
+                return "null"
+            }
+        case .ArrayOfStrings:
```
Comment (schmit) [inline Sources/eppo/dto/EppoValue.swift:182]: note: for subject attributes this is not allowed in other SDKs. I know you coalesced the types so up to you how you want to handle it here

File: Sources/eppo/dto/EppoValue.swift (hunk 2 above)
Comment (schmit) [reply Sources/eppo/dto/EppoValue.swift:194]: Not related to this PR, but wouldn't it make more sense to use the variation key instead of the variation value when hashing? Practically speaking wouldn't make a difference, but we avoid having to hash json blobs

File: Tests/eppo/EppoValueTests.swift (hunk 1 above)
Comment (schmit) [reply Tests/eppo/EppoValueTests.swift:71]: 👍 , we can always add more tests if we find more edge cases

**Review by schmit — APPROVED** [review]
🚀 so fast

File: Sources/eppo/dto/EppoValue.swift (hunk 3 above)
Comment (leoromanovsky) [reply Sources/eppo/dto/EppoValue.swift:171]: If the instance attribute doubleValue is null then it will throw; the type is a throwable double so I need to handle the failure case.

Practically this won't happen since we are in the code branch for doubles which matches the implementation above.

I can either return a sentinel value here or throw an exception, but that would make this function a throwable string.

File: Sources/eppo/dto/EppoValue.swift (hunk 2 above)
Comment (leoromanovsky) [reply Sources/eppo/dto/EppoValue.swift:194]: Yea good point I will review my old PR for why I decided to use this.
